src/Compilers/Core/Portable/Diagnostic/DiagnosticSeverity.cs

```diff
@@ -27,7 +27,12 @@ public enum DiagnosticSeverity
         /// <summary>
         /// Something not allowed by the rules of the language or other authority.
         /// </summary>
-        Error = 3
+        Error = 3,
+
+        /// <summary>
+        /// Something that can be potentially improved.
+        /// </summary>
+        Suggestion = 4,
```

Comment:
> not sure how Error+Unecessary make sense if we consider unnecessary as characteristic rather than visual effect.

> error can't be unnecessary. if we consider unnecessary as cue for visual effect than

Unnecessary _is_ a visual effect that has semantic meaning for the user. The user's experience is that their code is faded because it is unnecessary. How it impacts them from a severity standpoint (i.e. breaking the build or not) is orthogonal.

> hidden + suggestion also make sense. same as error, warning, info + suggestion.

I'd love it if you can expand on this, because it doesn't make any sense to me.